test/main/deepEqual.js

```diff
@@ -1528,3 +1528,226 @@ QUnit.test( "Test that must be done at the end because they extend some primitiv
 		);
 	}
 );
+
+QUnit.module( "equiv Maps and Sets");
+// Note: we cannot do simple check for constructor existence,
+// as IE 11 implementation of Sets/Maps is incomplete.
+//  - in IE 11, Sets/Maps do not support constructors, e.g. new Set( [1] ) fails.
+//  - in IE 11, QUnit.objectType( new Set() ) === 'object'
+//
+// This is a more robust way of checking for ES6 Set/Map support.
+var hasES6Set = (function() {
+	try {
+		var s = new Set([ 1, 2, 3 ]);
+		if (s.size !== 3 || !s.has(2)) {
+			return false;
+		}
+	}
+	catch (e) {
+		return false;
+	}
+
+	return true;
+})();
+
+var hasES6Map = (function() {
+	try {
+		var m = new Map([
+			[ 1, 2],
+			[ 3, 4]
+		]);
+		if (m.size !== 2 || !m.has(1)) {
+			return false;
+		}
+	}
+	catch (e) {
+		return false;
+	}
+
+	return true;
+})();
+
+QUnit[ hasES6Set ? "test" : "skip" ]("Sets", function ( assert ) {
+	var s1, s2, s3, s4, o1, o2, o3, m1, m2, m3;
+
+	// Empty sets
+	s1 = new Set();
+	s2 = new Set();
+	assert.equal( QUnit.equiv( s1, s2 ), true, "Empty Sets");
+	s1 = new Set([]);
+	s2 = new Set([]);
+	assert.equal( QUnit.equiv( s1, s2 ), true, "Empty sets init with empty arrays");
```

Comment:
`new Set();` and `new Set([]);`

Both returns empty Set objects, this is tested on ES6. We may replace this by only: 

```js
s1 = new Set();
s2 = new Set( [] );
assert.equal( QUnit.equiv( s1, s2 ), true, "Empty Sets" );
```